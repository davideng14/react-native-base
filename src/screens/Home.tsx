import React from 'react';
import { View, Button } from 'react-native';

export default function Home(props: any) {
    return (
    <View>
        <Button title={"Home2"} onPress={() => {
            props.navigation.navigate('StoreScreen');
        }} />
    </View>
    )
} 