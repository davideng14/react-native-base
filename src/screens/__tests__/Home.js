import 'react-native';
import React from 'react';
import Home from '../Home';
import { shallow } from 'enzyme';

import renderer from 'react-test-renderer';

it('renders correctly', () => { 
    const tree = renderer.create( 
      <Home /> 
    ); 
    expect(tree).toMatchSnapshot(); 
});

it('has proper navigation options', () => { 
    expect(Home.navigationOptions).toMatchSnapshot(); 
});

it('navigate on press', () => { 
    const navigate = jest.fn(); 
    const homeScreen = shallow( 
      <Home navigation={{ navigate }}/> 
    ) 
    homeScreen.find('Button').simulate('press'); 
    expect(navigate).toBeCalledWith("StoreScreen");
});