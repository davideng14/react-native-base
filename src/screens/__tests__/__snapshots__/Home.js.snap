// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`has proper navigation options 1`] = `undefined`;

exports[`renders correctly 1`] = `
<View>
  <View
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Home2
      </Text>
    </View>
  </View>
</View>
`;
